### DISCUSSION TITLE
No way to find the author of an item when using pods_api_pre_save_pod_item

### DISCUSSION THREAD

**Discussion by lougreenwood** [pods-framework/pods#6642] [Q&A]
It seems that when using the `pods_api_pre_save_pod_item` filter, there's no way to know the author of the item about to be saved.

In my use case, `wp_get_current_user` can't be used, because the current user may be an admin creating a record for someone else.

**Comment by lougreenwood** [comment]
Ok, seems this is possible using some un-documented steps:

set number of params for `pods_api_pre_save_pod_item` add action to `3` (docs state 2).

This provides the post ID of the item to be saved (presumably auto-draft....?).

In the handler method for the filter, do `$author_id = (int) get_post_field( 'post_author', $id );` where `$id` is the third param provided by by `pods_api_pre_save_pod_item`.

Perhaps the docs need updating with this 3rd param?

**Comment by quasel** [comment]
hi take a look at 
https://docs.pods.io/code-snippets/create-post-title-from-fields-in-the-post-using-pods_api_pre_save/

the author is part of $pieces[ 'object_fields' ] like the post title - sorry for the trouble :(  

@jimtrue maybe a thing for updating docs!

**Comment by jimtrue** [comment]
@quasel I'd say on these what is _more_ helpful? We can't possibly document every single thing in $pieces because it's a huge structure. We could give folks a 'road map' to identify where to find things they more than likely would be using.

@sc0ttkclark Does that sound like something you could easily do?
